earth_enterprise/src/fusion/gepublish/CurlRequest.h

```diff
@@ -26,7 +26,106 @@
 
 #include "common/ManifestEntry.h"
 
-class geProgress;
+class geProgress; 
+class GetRequest;
+class UploadRequest;
+
+typedef size_t (*CurlRequestWriteFunc)(void*, size_t, size_t, void*);
+typedef size_t (*GetRequestWriteFunc)(void*, size_t, size_t, GetRequest*);
+typedef size_t (*UploadRequestReadFunc)(void*, size_t, size_t, UploadRequest*);
+
+struct curl_api {
+  virtual CURL* easy_init() = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, bool) = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, int) = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, long) = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, unsigned long) = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, const char*) = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, curl_slist*) = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, CurlRequestWriteFunc) = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, GetRequestWriteFunc) = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, UploadRequestReadFunc) = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, std::nullptr_t) = 0;
+  virtual CURLcode easy_setopt(CURL *handle, CURLoption option, void*) = 0;
+  virtual curl_slist*slist_append(curl_slist* lst, const char * item) = 0;
+  virtual void slist_free_all(curl_slist* lst) = 0;
+  virtual void easy_cleanup(CURL *handle) = 0;
+  virtual CURLcode easy_perform(CURL * handle) = 0;
+  virtual CURLcode easy_getinfo(CURL *curl, CURLINFO info, long* result_code) = 0;
+};
```

Comment:
Added this to make the request classes testable without having to connect to a real rest server.  This will let you fake out curl calls during testing.